### PR TITLE
allow special entities to be defined what operations they support and allow them to be extendable

### PR DESCRIPTION
**Pull request by dpfaffenbauer**

### CODE REVIEW

File: src/Controller/ConfigController.php [hunk 1]
```diff
@@ -341,14 +345,79 @@ public function getAction(Request $request, Service $graphQlService): JsonRespon
...
+                $config['schema']['specialEntities']['asset_folder']['update'] ?? false,
+                $config['schema']['specialEntities']['asset_folder']['delete'] ?? false
+            ),
+            new SpecialEntitySetting(
+                'asset_folder',
+                true,
+                true,
+                true,
+                true,
+                $config['schema']['specialEntities']['asset_listing']['read'] ?? false,
```
Comment (weisswurstkanone) [inline src/Controller/ConfigController.php:399]: asset_folder or asset_listing ?

File: src/Controller/ConfigController.php [hunk 2]
```diff
@@ -341,14 +345,79 @@ public function getAction(Request $request, Service $graphQlService): JsonRespon
...
+                $config['schema']['specialEntities']['asset_listing']['update'] ?? false,
+                $config['schema']['specialEntities']['asset_listing']['delete'] ?? false
+            ),
+            new SpecialEntitySetting(
+                'asset_listing',
+                true,
+                false,
+                false,
+                false,
+                $config['schema']['specialEntities']['object_folder']['read'] ?? false,
```
Comment (weisswurstkanone) [inline src/Controller/ConfigController.php:410]: object_folder or asset_listing ?

File: src/Controller/ConfigController.php (hunk 1 above)
Comment (dpfaffenbauer) [reply src/Controller/ConfigController.php:399]: fixed

File: src/Controller/ConfigController.php (hunk 2 above)
Comment (dpfaffenbauer) [reply src/Controller/ConfigController.php:410]: fixed

File: src/Controller/ConfigController.php [hunk 3]
```diff
@@ -341,14 +345,79 @@ public function getAction(Request $request, Service $graphQlService): JsonRespon
...
+                true,
+                true,
+                $config['schema']['specialEntities']['asset_listing']['read'] ?? false,
+                $config['schema']['specialEntities']['asset_listing']['create'] ?? false,
+                $config['schema']['specialEntities']['asset_listing']['update'] ?? false,
+                $config['schema']['specialEntities']['asset_listing']['delete'] ?? false
+            ),
+            new SpecialEntitySetting(
+                'object_folder',
+                true,
```
Comment (weisswurstkanone) [inline src/Controller/ConfigController.php:406]: Could you please recheck the matrix? I don't see why CUD should not be possible for object folders.

Comment (dpfaffenbauer) [reply src/Controller/ConfigController.php:406]: done

**Comment by weisswurstkanone** [comment]
How did this happen? Also the permissions are gone.

image

**Comment by weisswurstkanone** [comment]
seems that something is wrong here.
on the left side you see my previous configuration file, on the right side the new one. why would 

image

**Comment by dpfaffenbauer** [comment]
fixed

File: src/Controller/ConfigController.php [hunk 4]
```diff
@@ -319,14 +323,85 @@ public function getAction(Request $request, Service $graphQlService): JsonRespon
...
+                true,
+                $config['schema']['specialEntities']['asset_folder']['read'] ?? false,
+                $config['schema']['specialEntities']['asset_folder']['create'] ?? false,
+                $config['schema']['specialEntities']['asset_folder']['update'] ?? false,
+                $config['schema']['specialEntities']['asset_folder']['delete'] ?? false
+            ),
+            new SpecialEntitySetting(
+                'asset_listing',
+                true,
```
Comment (weisswurstkanone) [inline src/Controller/ConfigController.php:373]: create update delete is possible for listings ?

**Comment by weisswurstkanone** [comment]
maybe also mention this feature in the docs

File: src/Controller/ConfigController.php (hunk 4 above)
Comment (dpfaffenbauer) [reply src/Controller/ConfigController.php:373]: well, actually not, but that wasn't limited before this PR also.

**Comment by dpfaffenbauer** [comment]
oops, forgot about that, will try to change docs tomorrow

**Comment by dpfaffenbauer** [comment]
done

**Comment by weisswurstkanone** [comment]
thanks a lot!

**Pull request merged**